source/common/umf_pools/disjoint_pool.cpp

```diff
@@ -293,6 +293,16 @@ class DisjointPool::AllocImpl {
         // Generate buckets sized such as: 64, 96, 128, 192, ..., CutOff.
         // Powers of 2 and the value halfway between the powers of 2.
         auto Size1 = params.MinBucketSize;
+        // Special case for when MinBucketSize equals 1 because otherwise bucket sizes starting from 1 value
+        // would generate with duplicates (like 1,1,2,2,4,4...) which is undesirable.
```

Comment:
I think this statement is misleading: `(like 1,1,2,2,4,4...)`. Only `1` will be duplicated. If the `Size1` equals `2` the sequence without this patch is `2, 3, 4, 6, 8, 12, 16, 24, etc`

Reply:
Yeah, I think you can just write that we only generate powers of 2 up to a certain value (64) and then powers of 2 + values in between.